### PR TITLE
fix: compose dropping `only.localOS` in an only filter

### LINKED ISSUE #2158: Zarf 0.31.1: "localOS: ..." Tag in zarf.yaml does not work anymore (on Windows and Linux)
### Environment
Device and OS: Linux (and Windows Server 2019)
App version: Zarf v0.31.1
Kubernetes distro being used: k3s with zarf included (running only on Linux)
Other:

### Steps to reproduce
1. Build Zarf Package with Zarf v.0.31.1 on Linux
2. Deploy the same Zarf Package on a Linux system with atleast following zarf.yaml (should be in build too):

```yaml
  - name: basic name
    only:
      localOS: windows
    actions:
      onDeploy:
        after:
          - cmd: "<Placeholder for Powershell-only Command>"
```

3. Build Zarf Package with Zarf v.0.31.1 on Linux
4. Deploy the same Zarf Package on a Windows system with atleast following zarf.yaml (should be in build too):

```yaml
  - name: basic name
    only:
      localOS: linux
    actions:
      onDeploy:
        after:
          - cmd: "<Placeholder for bash/linux-only Command>"
```

### Expected result

- When the first yaml-block (localOS: windows) is run on a Linux System, it should not be interpreted at all by bourneshell and just skipped. When its run on a Windows System, it should be interpreted and run correctly.

- When the seconf yaml-block (localOS: Linux) is run on a Windows System, it should not be interpreted at all by Powershell and just skipped. When its run on a Linux System, it should be interpreted and run correctly.

### Actual Result

- When `zarf deploy <packagename>` is executed on a linux system, there is an error: `sh: 1: Syntax error: <powershell-command, that cant get interpreted by bourneshell>`

- When `zarf deploy <packagename>` is executed on a windows system, there is an error: `Powershell error: <Bash-command that cant get interpreted by powershell>`

### Visual Proof (screenshots, videos, text, etc)
zarf.yaml:
errorforzarfissue
Powershell-Error:
errorpowershell
Linux-Error:
yamlerrorlinux

### Severity/Priority

- Medium, because the functionality above is not working anymore

### Additional Context
I upgraded from Zarf Version v0.29.2 to v0.31.1. In Version v0.29.2 it worked perfectly fine. I didn't alter the code in any shape or form.

### PR DESCRIPTION
**Pull request by Racer159**
## Description

This fixes an issue where compose could drop the `only.localOS` filter through compose.

## Related Issue

Fixes #2158

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [X] Test, docs, adr added or updated as needed
- [X] [Contributor Guide Steps](https://github.com/defenseunicorns/zarf/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

**Review by lucasrod16 — COMMENTED** [review]
We need to dereference the "composed" pointer on line 47 in `src/pkg/packager/compose.go`:

components = append(components, composed) -> components = append(components, *composed)

File: src/pkg/packager/composer/list.go [hunk 1]
```diff
@@ -229,20 +229,24 @@ func (ic *ImportChain) Compose() (composed types.ZarfComponent, err error) {
...
 		fixPaths(&node.ZarfComponent, node.relativeToHead)
 
 		// perform overrides here
-		overrideMetadata(&composed, node.ZarfComponent)
-		overrideDeprecated(&composed, node.ZarfComponent)
-		overrideResources(&composed, node.ZarfComponent)
-		overrideActions(&composed, node.ZarfComponent)
+		err := overrideMetadata(composed, node.ZarfComponent)
+		if err != nil {
+			return composed, err
```
Comment (lucasrod16) [inline src/pkg/packager/composer/list.go:242]: Would probably be good to update all error handling in `Compose()` to return a nil pointer

```suggestion
			return nil, err
```

**Comment by Noxsios** [comment]
> We need to dereference the "composed" pointer on line 47 in `src/pkg/packager/compose.go`:
> 
> `components = append(components, composed)` -> `components = append(components, *composed)`

relevant screenshot:

image

**Review by Noxsios — APPROVED** [review]

File: src/pkg/packager/composer/list.go
Comment (lucasrod16) [inline src/pkg/packager/composer/list.go:1]: Line 228:

`return composed, err` -> `return nil, err`

**Review by lucasrod16 — CHANGES_REQUESTED** [review]

File: src/pkg/packager/composer/list.go
Comment (Racer159) [reply src/pkg/packager/composer/list.go:1]: done

**Review by lucasrod16 — APPROVED** [review]
lgtm

**Review by Noxsios — APPROVED** [review]

**Pull request merged**